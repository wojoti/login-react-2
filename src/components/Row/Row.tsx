import React, { ReactNode } from "react";
import { Wrapper } from "./Row.style";

type Props = {
  children: ReactNode;
};

const Row = (props: Props) => {
  return <Wrapper>{props.children}</Wrapper>;
};

export default Row;
