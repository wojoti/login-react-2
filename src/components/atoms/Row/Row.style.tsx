import styled from "styled-components";

interface Props {
  mt?: number;
  mb?: number;
  ml?: number;
  mr?: number;
  pt?: number;
  pb?: number;
  pl?: number;
  pr?: number;
  justify?: string;
  align?: string;
}

const Wrapper = styled.div<Props>`
  box-sizing: border-box;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;

  margin-top: ${(props) => props.mt || 0}px;
  margin-bottom: ${(props) => props.mb || 0}px;
  margin-left: ${(props) => props.ml || 0}px;
  margin-right: ${(props) => props.mr || 0}px;
  padding-top: ${(props) => props.pt || 0}px;
  padding-bottom: ${(props) => props.pb || 0}px;
  padding-left: ${(props) => props.pl || 0}px;
  padding-right: ${(props) => props.pr || 0}px;
  display: flex;
  flex-direction: row;
  justify-content: ${(props) => props.justify || "start"};
  align-items: ${(props) => props.align || "start"};
`;
export default Wrapper;
