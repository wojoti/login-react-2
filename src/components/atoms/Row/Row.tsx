import Wrapper from "./Row.style";

export interface RowProps {
  children?: React.ReactNode;
  mt?: number;
  mb?: number;
  ml?: number;
  mr?: number;
  pt?: number;
  pb?: number;
  pl?: number;
  pr?: number;
  justify?:
    | "flex-start"
    | "center"
    | "flex-end"
    | "space-between"
    | "space-around"
    | "space-evenly";
  align?:
    | "flex-start"
    | "center"
    | "flex-end"
    | "space-between"
    | "space-around"
    | "space-evenly";
  testId?: string;
}

function Row({
  mt,
  mb,
  ml,
  mr,
  pt,
  pb,
  pl,
  pr,
  justify,
  align,
  testId,
  children,
}: RowProps) {
  return (
    <Wrapper
      mt={mt}
      mb={mb}
      ml={ml}
      mr={mr}
      pt={pt}
      pb={pb}
      pl={pl}
      pr={pr}
      justify={justify}
      align={align}
      data-testid={testId}
    >
      {children}
    </Wrapper>
  );
}

export default Row;
