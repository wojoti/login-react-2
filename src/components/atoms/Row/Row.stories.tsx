/* eslint-disable react/function-component-definition */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-extraneous-dependencies */
import { ComponentMeta, ComponentStory } from "@storybook/react";

import Row from "./Row";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "Atoms/Row",
  component: Row,
  decorators: [
    (Story) => (
      <div style={{ border: "1px solid lightgray" }}>
        <Story />
      </div>
    ),
  ],
} as ComponentMeta<typeof Row>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Row> = (args) => (
  <Row {...args}>
    <span>Example span.</span>
    <span>Example span.</span>
    <span>Example span.</span>
  </Row>
);

export const Default = Template.bind({});
Default.args = {};
