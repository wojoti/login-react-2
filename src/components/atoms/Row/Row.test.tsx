import { render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import Row, { RowProps } from "./Row";

const props: RowProps = {
  testId: "test-row-id",
};

const modifiedProps: RowProps = {
  ...props,
  mr: 14,
};

test("should match snapshot", () => {
  const tree = TestRenderer.create(<Row {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render row", () => {
  render(<Row {...props} />);
  const rowElement = screen.getByTestId("test-row-id");
  expect(rowElement).toBeInTheDocument();
});

test("should render row with modified style through props", () => {
  render(<Row {...modifiedProps} />);
  const rowElement = screen.getByTestId("test-row-id");
  expect(rowElement).toBeInTheDocument();
  expect(rowElement).toHaveStyle("margin-right: 14px");
});
