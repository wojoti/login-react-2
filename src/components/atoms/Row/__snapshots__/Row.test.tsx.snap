// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  margin-top: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
  margin-right: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

<div
  className="c0"
  data-testid="test-row-id"
/>
`;
